File: go-sdk/pkg/models/operations/deleteweblinksidtrash.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type DeleteWebLinksIDTrashPathParams struct {
	WebLinkID string `pathParam:"style=simple,explode=false,name=web_link_id"`
}

type DeleteWebLinksIDTrashRequest struct {
	Retries    *utils.RetryConfig
	PathParams DeleteWebLinksIDTrashPathParams
}

type DeleteWebLinksIDTrashResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
}

File: ts-sdk/src/sdk/models/operations/deleteweblinksidtrash.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class DeleteWebLinksIdTrashPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=web_link_id" })
  webLinkId: string;
}


export class DeleteWebLinksIdTrashRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: DeleteWebLinksIdTrashPathParams;
}


export class DeleteWebLinksIdTrashResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
